src/main/java/com/werfen/report/service/ExcelGridReportService.java

```diff
@@ -0,0 +1,157 @@
+    }
+
+
+
+    private void createSummary(GridReportConfiguration gridReportConfiguration, Workbook workbook)  {
+        try {
+            ReportHeaderConfiguration headerConfiguration = gridReportConfiguration.getHeaderConfiguration();
+            ReportFooterConfiguration reportFooterConfiguration = gridReportConfiguration.getFooterConfiguration();
+            Sheet sheet = workbook.createSheet("Summary");
+            Row header = sheet.createRow(0);
+            Cell headerCell  = header.createCell(0) ;
+            headerCell.setCellValue(headerConfiguration.getTitle());
+            FileInputStream inputStream = new FileInputStream( headerConfiguration.getLogoPath() );
+
+            int pictureIndex = workbook.addPicture(IOUtils.toByteArray(inputStream), Workbook.PICTURE_TYPE_PNG);
+            inputStream.close();
+
+            CreationHelper helper = workbook.getCreationHelper();
+            Drawing drawing = sheet.createDrawingPatriarch();
+            ClientAnchor anchor = helper.createClientAnchor();
+            anchor.setCol1(2);
+            anchor.setRow1(0);
+            anchor.setCol2(4);
+            anchor.setRow2(1);
+            Picture pict = drawing.createPicture(anchor, pictureIndex);
+            header.createCell(2) ;
+
+            int row = 2;
+
+
+            if (nonNull(headerConfiguration.getField1())) {
+                createHeaderFieldsData(headerConfiguration.getField1(), sheet, row);
+                row ++;
+            }
+            if (nonNull(headerConfiguration.getField2())) {
+                createHeaderFieldsData(headerConfiguration.getField2(), sheet, row);
+                row++;
+            }
+            if (nonNull(headerConfiguration.getField3())) {
+                createHeaderFieldsData(headerConfiguration.getField3(), sheet, row);
+                row ++;
+            }
+            if (nonNull(headerConfiguration.getField4())) {
+                createHeaderFieldsData(headerConfiguration.getField4(), sheet, row);
+                row ++;
+            }
+            row = row + 2;
+
+            if (nonNull(reportFooterConfiguration.getField1())) {
+                createHeaderFieldsData(reportFooterConfiguration.getField1(), sheet, row);
+                row++;
+            }
+            if (nonNull(reportFooterConfiguration.getField2())) {
+                createHeaderFieldsData(reportFooterConfiguration.getField2(), sheet, row);
+                row++;
+            }
+            if (nonNull(reportFooterConfiguration.getField3())) {
+                createHeaderFieldsData(reportFooterConfiguration.getField3(), sheet, row);
+                row++;
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void createHeaderFieldsData(GridReportField gridReportField, Sheet sheet, int row) {
```

Comment:
If we go for the approach of renaming methods of this class to start with "add", I think that the first argument should be the container where we are adding things into. In this case, it corresponds with the POI entity (sheet or workbook). Again, this kind of practices facilitate reading the code.